squad_client/commands/submit_tuxbuild.py

```diff
@@ -60,7 +62,46 @@
     }],
 }
 
+
+def load_builds(build_json):
+    try:
+        with open(build_json) as f:
+            return json.load(f)
+    except json.JSONDecodeError as jde:
+        raise InvalidBuildJson(f"Invalid build json: {jde}")
+
+
+def create_name(build):
+    suite = "build/"
+    name = ""
+
+    if build["build_name"]:
+        name = build["build_name"]
+    else:
+        name += "%s-%s-%s" % (
+            build["target_arch"],
+            build["toolchain"],
+            build["kconfig"][0],
+        )
+
+        if len(build["kconfig"]) > 1:
+            name += "-" + create_sha(build)
+
+    return suite + name
+
+
+def create_sha(build):
+    sha = hashlib.sha1()
+
+    # log?
+    for k in build["kconfig"][1:]:
+        sha.update(f"{k}".encode())
+
+    return sha.hexdigest()[0:8]
+
+
 ALLOWED_METADATA = [
+    "config",
```

Comment:
why do we need this?

Reply:
It should of been added in a previous commit, but I missed it in a review. I added it in to avoid some test failures.

https://github.com/Linaro/squad-client/blame/06612961fc16c5f4af9983fcfe6f378e7159f965/squad_client/commands/submit_tuxbuild.py#L113

Reply:
Let me know if that does not make sense!

Reply:
I don't think it make sense to add it, no.
I saw the ALLOWED_METADATA as a filter for what will be added from the build.json file that are submitted. Not if we add extra fields after that.